Assets/_BuildingBlocks/Credentials/Scripts/KeyVault.cs

```diff
@@ -10,317 +10,212 @@
 
-            StoredAuthorization.StoredAuthorization auth = new FailedOrUnsupported(uri);
-            switch (authorizationType)
+            foreach (var supportedAuthTypes in supportedAuthorizationTypes)
             {
-                case AuthorizationType.Public:
-                    auth = new Public(uri);
-                    storedAuthorizations.Add(baseUri, auth);
-                    break;
-                case AuthorizationType.UsernamePassword:
-                    auth = new UsernamePassword(uri, username, password);
-                    storedAuthorizations.Add(baseUri, auth);
-                    break;
-                // case AuthorizationType.InferableSingleKey:
-                    // auth = new QueryStringAuthorization(uri, key);
-                    // storedAuthorizations.Add(baseUri, auth);
-                    // break;
-                case AuthorizationType.Key:
-                    auth = new Key(uri, key);
-                    storedAuthorizations.Add(baseUri, auth);
-                    break;
-                case AuthorizationType.BearerToken:
-                    auth = new BearerToken(uri, key);
-                    storedAuthorizations.Add(baseUri, auth);
-                    break;
-                case AuthorizationType.Code:
-                    auth = new Code(uri, key);
-                    storedAuthorizations.Add(baseUri, auth);
-                    break;
-                case AuthorizationType.Token:
-                    auth = new Token(uri, key);
-                    storedAuthorizations.Add(baseUri, auth);
-                    break;
-                case AuthorizationType.FailedOrUnsupported:
-                    // we don't want to save a failed auth, since this could be a typo and it should not automatically fail if the user retries.
-                    // If possible we want to make a distinction between Failed and Unsupported, and only save if it's unsupported, since then retrying is futile.
-                    break;
-                default:
-                    throw new ArgumentOutOfRangeException(nameof(authorizationType), authorizationType, "authorisation type not defined");
-            }
-
-            OnAuthorizationTypeDetermined.Invoke(auth);
-        }
-
-        public void ClearURLFromStoredAuthorizations(Uri url)
-        { 
-            storedAuthorizations.Remove(url);
-        }
-
-        /// <summary>
-        /// Try to find the specific type of credential (key, token or code) that is needed for the layer
-        /// </summary>
-        public void TryToFindSpecificCredentialType(Uri uri, string key)
-        {
-            //Only allow one simultaneous coroutine for now
-            if (coroutineMonoBehaviour != null)
-                Destroy(coroutineMonoBehaviour.gameObject);
+                if (!supportedAuthTypes.Key.IsSubclassOf(typeof(QueryStringAuthorization))) //only check our supported auth types that could have the auth in the url.
```

Comment:
had dit niet een `supportedAuthTypes.Key is not QueryStringAuthorization` kunnen zijn?

Reply:
nee, want storedAuthorizationType is hier een `Type` en niet een StoredAuthorization